### ISSUE TITLE
Can't understand what is wrong in POST request

### ISSUE THREAD

**Issue by revverse** [AppsFlyer/go-sundheit#20]
I used example in README like this:

``` package main

import (
  "bytes"
  "fmt"
  health "github.com/AppsFlyer/go-sundheit"
  //"io/ioutil"
  "net/http"
  "time"
  "log"

  healthhttp "github.com/AppsFlyer/go-sundheit/http"
  "github.com/AppsFlyer/go-sundheit/checks"
)

func main() {
  // create a new health instance
  h := health.New()
  s := '{"test":"json"}'
  r := bytes.NewBuffer([]byte(s))
  httpCheckConf := checks.HTTPCheckConfig{
    CheckName: "reverse",
    Timeout:   1 * time.Second,
    Body:	r,
    // dependency you're checking - use your own URL here...
    // this URL will fail 50% of the times
    URL:       "http://127.0.0.1:9991/g",
  }
  // create the HTTP check for the dependency
  // fail fast when you misconfigured the URL. Don't ignore errors!!!
  httpCheck, err := checks.NewHTTPCheck(httpCheckConf)
  if err != nil {
    fmt.Println(err)
    return // your call...
  }

  // Alternatively panic when creating a check fails
  httpCheck = checks.Must(checks.NewHTTPCheck(httpCheckConf))

  err = h.RegisterCheck(&health.Config{
    Check:           httpCheck, 
    InitialDelay:    time.Second,      // the check will run once after 1 sec
    ExecutionPeriod: 10 * time.Second, // the check will be executed every 10 sec
  })
  
  if err != nil {
    fmt.Println("Failed to register check: ", err)
    return // or whatever
  }

  // define more checks...
  
  // register a health endpoint
  http.Handle("/admin/health.json", healthhttp.HandleHealthJSON(h))
  
  // serve HTTP
  log.Fatal(http.ListenAndServe(":8080", nil))
}
```

In this code we POST simple JSON 
`  s := '{"test":"json"}'
  r := bytes.NewBuffer([]byte(s))
`

When app started everything is OK - request sent as expected. But on the second iteration i have empty Body in request because they have type io.Reader and can`t use second time.
tcpdump:

```
GET /g HTTP/1.1
Host: 127.0.0.1:9991
User-Agent: Go-http-client/1.1
Content-Length: 15
Accept-Encoding: gzip

{"test":"json"}HTTP/1.1 404 Not Found
Content-Type: text/plain
Date: Wed, 29 Apr 2020 13:46:40 GMT
Content-Length: 18

404 page not found
-----

GET /g HTTP/1.1
Host: 127.0.0.1:9991
User-Agent: Go-http-client/1.1
Accept-Encoding: gzip

HTTP/1.1 404 Not Found
Content-Type: text/plain
Date: Wed, 29 Apr 2020 12:36:04 GMT
Content-Length: 18

404 page not found
```

I am newby in Golang and maybe don't understand how to send Body on each check.
In my fork i change type of Body to 'string' and write  NewRequest via NewReader:

```
- req, err := http.NewRequest(check.config.Method, check.config.URL, check.config.Body)
+ req, err := http.NewRequest(check.config.Method, check.config.URL, bytes.NewReader([]byte(check.config.Body)))
```

After that example send Body on each ticker
diff https://github.com/AppsFlyer/go-sundheit/compare/master...revverse:master

**Comment by eranharel** [comment]
I'm sorry, I don't understand what you're trying to do, nor what's not working exactly.
It seems like you're asking a general go HTTP client how-to question, but your code examples are partial so it's hard to tell.

I suggest that you either
- Complete the missing parts of your question, including relevant code snippets.
- Try to call the service health endpoint from `curl` first, and only then try to compose the golang code
- Try to find help on StackOverflow or the go docs for the code that's not working

Without additional inputs I can't really help, sorry

**Comment by revverse** [comment]
I try to say that, when i create simple http checker with defined **Body** (full code in first message) , application started and send http body to endpoint only on first check, second check will send empty body

**Comment by eranharel** [comment]
@revverse in the code you posted, you register a single HTTP check, then register the health endpoint handler under `/admin/health.json` and start the server in port 8080.

The get request you posted after looks totally unrelated...
```
GET /g HTTP/1.1
Host: 127.0.0.1:9991
```

You also didn't really use those strings:
```
  s := '{"test":"json"}'
  r := bytes.NewBuffer([]byte(s))
```

**Comment by revverse** [comment]
Yes application started on localhost:8080, but checkURL is:
```    Body:	r,
    URL:       "http://127.0.0.1:9991/g",
```
port 9991 listen by 'some server' that answers 200OK only on '/'  , on other endpoints - 404. It's only for test. But problem in the second request generated by application (Checker):
```
GET /g HTTP/1.1
Host: 127.0.0.1:9991\
.....
{"test":"json"}

HTTP/1.1 404 Not Found

GET /g HTTP/1.1
Host: 127.0.0.1:9991

HTTP/1.1 404 Not Found
 ```
Body is **empty** . It's not depends on send method. I tried this:
```
      Method:          "POST",
      ExpectedStatus:  404,
```
But again - empty body in request

**Comment by eranharel** [comment]
I think I understand the problem now.
Let me see what I can do after the weekend

**Comment by eranharel** [comment]
@revverse want to try this out before I merge? (feel free to review too)
https://github.com/AppsFlyer/go-sundheit/pull/21

**Comment by revverse** [comment]
With this PR, check.Body present in every check. Thanks! 
And sorry for my poor English

**Comment by revverse** [comment]
Only one note - when we try to check our config via:
```
  // Alternatively panic when creating a check fails
  httpCheck = checks.Must(checks.NewHTTPCheck(httpCheckConf))
```
**payload** will be empty because we use ` ioutil.ReadAll(config.Body)` again in  NewHTTPCheck
But it's not a problem to me . Thanks

**Comment by eranharel** [comment]
@revverse I released `v0.2.0`. 
It is a slightly breaking release, as I made the `HTTPCheckConfig.Body` a function.
It's just a function that will return a new Reader every time, and it will allow you to vary the payload, and to also post a large file if needed
